### ISSUE TITLE
On Firefox, scrolling keys don't work when input is focused

### ISSUE THREAD

**Issue by JoshuaGrams** [erkyrath/glkote#81]
At least Firefox 131.0.3 on Windows, for me.

It's not clear to me yet why this is happening.

This [code in `evhan_doc_keypress`](https://github.com/erkyrath/glkote/blob/master/glkote.js#L2634) talks about scrolling keys, but AFAICT it's both dead code and wrong code because 1) it's in a keypress event, which doesn't get non-printable key events (and I think never has? IIRC that was always one of its main weaknesses), and 2) it treats keycodes as if they're ASCII, which they aren't entirely (the logic here for "is this a printable key" is `keycode >= 32 && keycode <= 126` but PageUp, PageDown, End, and Home are 33-36, respectively).

From messing around in the debugger it looks like these keys only pass through `evhan_input_keydown` and then the scrolling happens (on Chrome) after that event handler returns but I'm not yet sure why it doesn't happen on Firefox. I'll keep looking, but I wanted to file an issue in case it's an obvious fix for someone else...

**Comment by erkyrath** [comment]
Ah, this is what we were chatting about last night.

Thanks for digging into it. Yes, I see the same problem on Mac Firefox.

**Comment by erkyrath** [comment]
You're right about the dead bit of code. I've deleted that. But it was inside the `if (windows_paging_count)` case -- that is, when in MORE paging mode. So it wouldn't affect your situation even if it was corrected.

In evhan_input_keydown(), returning false on 33/34 _prevents_ Chrome/Safari from scrolling the pane.

I guess one solution is to scroll manually (calling scrollTop()) and then return false? That would give consistent results on all browsers. But it would look a little weird, in that pageup/down would be smooth scrolling (default browser behavior) when the input was not focused, but instant (calling scrollTop()) when the input _was_ focused.

Still, probably better than the current situation.
